src/Microsoft.OData.Core/Json/JsonReader.cs

```diff
@@ -382,6 +382,19 @@ public virtual bool Read()
         }
 
         /// <summary>
+        /// Extracts text before and including the current token for use in an error message.
+        /// </summary>
+        /// <param name="maxContextLength">The maximum number of characters to include in the parsing context.</param>
+        /// <returns>Text from the character buffer.</returns>
+        internal string CaptureParsingContext(int maxContextLength = 100)

```

Comment:
It is parameterized mainly for testing. I chose 100 based on some real-world testing with my OData-based service (Azure Search). 100 characters gives enough context to identify the invalid JSON without making the error message too large. If you disagree, we can change it.